packages/core/src/components/essentials/legend.ts

```diff
@@ -268,6 +268,7 @@ export class Legend extends Component {
 				const clickedItemData = clickedItem.datum() as any;
 
 				self.model.toggleDataLabel(clickedItemData.name);
+				self.model.updateSelectedGroups();
```

Comment:
not sure how this needs to happen, and I'm not sure how realistic it is for the dev to pass in `selectedGroups`. what if the chart updates? or the user interacts with the legend?

Reply:
> not sure how this needs to happen, and I'm not sure how realistic it is for the dev to pass in `selectedGroups`. what if the chart updates? or the user interacts with the legend?

When user check the legend, this will add checked legendItems to selectedGroup, if no legend is checked which means the chart will show all the data groups, then selectedGroup will be updated to an empty array since no legend is pre-selected. I think maybe pre-selectedGroup will be more decent for this option. What do you think ?

And chart updates you mean the data has changed ?

Reply:
yes I'm concerned with any updates really